packages/client-python/src/rocketride/_engine/resolver.py

```
@@ -0,0 +1,94 @@
+    Raises EngineNotFoundError if no compatible release is found.
+    """
+    if compat_range is None:
+        compat_range = get_compat_range()
+
+    spec = SpecifierSet(compat_range)
+
+    async with aiohttp.ClientSession() as session:
+        async with session.get(
+            _GITHUB_API,
+            headers={'Accept': 'application/vnd.github+json'},
+        ) as resp:
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)